modules/apps/frontend-js/frontend-js-item-selector-web/src/main/resources/META-INF/resources/item_selector/itemSelectorModal.tsx

```diff
@@ -80,33 +83,41 @@ function ItemSelectorModal<T extends Record<string, any>>({
 			<ClayModal.Body className="p-0">
 				<FrontendDataSet
 					{...fdsProps}
-					onSelect={({
-						selectedItems,
-					}: {
-						selectedItems: Array<T> | T;
-					}) => {
-						fdsProps.selectionType === 'single'
-							? setSelectedItem((selectedItems as Array<T>)[0])
-							: setSelectedItem(selectedItems as T);
-					}}
+					onSelectedItemsChange={setSelectedItems}
 					style="fluid"
 				/>
 			</ClayModal.Body>
 
 			<ClayModal.Footer
 				className={classNames({
-					'bg-primary-l3 border-primary border-top': selectedItem,
+					'bg-primary-l3 border-primary border-top': hasSelectedItems,
 				})}
 				first={
-					selectedItem ? (
-						<>
+					hasSelectedItems ? (
+						<div className="align-items-center d-flex">
 							{sub(
 								Liferay.Language.get('x-selected'),
-								<strong>
-									{getSelectedItemName(selectedItem)}
-								</strong>
+								selectedItems.length === 1
+									? getSelectedItemName(selectedItems[0])
+									: selectedItems.length
+							)}
+
+							{selectedItems.length > 1 && (
```

Comment:
After discussing this with design we should always show the clear button whenever any items are selected.